sru/release-27.14/test-migrate-user-config.sh

```diff
@@ -0,0 +1,206 @@
+
+    upgrade_to_proposed $name "NO CONFFILE PROMPT"
+
+    echo -e "\n-------------------------------------------"
+    echo "** Show uaclient.conf"
+    echo "-------------------------------------------"
+    lxc exec $name -- cat /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show user config"
+    echo "-------------------------------------------"
+    lxc exec $name -- pro config show
+    echo "-------------------------------------------"
+
+    lxc delete --force $name
+    echo "###########################################"
+}
+
+function test_uaclient_conf_changes_upgrade {
+    series=$1
+    old_version=$2
+    echo -e "\n\n###########################################"
+    echo "## $series: $old_version -> 27.13.4: preserve uaclient.conf changes"
+    echo "###########################################"
+    name=$(echo $series-$old_version | tr .~ -)
+
+    echo -e "\n-------------------------------------------"
+    echo "** launching container"
+    echo "-------------------------------------------"
+    lxc launch -q ubuntu-daily:$series $name
+    sleep 5
+    lxc exec $name -- apt-get update > /dev/null
+    lxc exec $name -- apt-get install debsums -y > /dev/null
+    echo "-------------------------------------------"
+
+    install_old_version $name $series $old_version
+
+    echo -e "\n-------------------------------------------"
+    echo "** make changes to uaclient.conf root"
+    echo "-------------------------------------------"
+    lxc exec $name -- sed -i "s/debug/warning/" /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show uaclient.conf"
+    echo "-------------------------------------------"
+    lxc exec $name -- cat /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show user config"
+    echo "-------------------------------------------"
+    lxc exec $name -- pro config show
+    echo "-------------------------------------------"
+
+    upgrade_to_proposed $name "NO CONFFILE PROMPT"
+
+    echo -e "\n-------------------------------------------"
+    echo "** Show uaclient.conf"
+    echo "-------------------------------------------"
+    lxc exec $name -- cat /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show user config"
+    echo "-------------------------------------------"
+    lxc exec $name -- pro config show
+    echo "-------------------------------------------"
+
+    lxc delete --force $name
+    echo "###########################################"
+}
+
+
+# xenial
+test_normal_upgrade         xenial 27.11.3~16.04.1
```

Comment:
On each test, should we also show the output of  `cat /var/lib/ubuntu-advantage/user-config.json`  ? II know we are already doing that through `pro config show`, but is just to make it explicit that the file doesn't contain any extra information we don't need.

Additionally, should we check that we have removed `/etc/ubuntu-advantage/uaclient.conf.preinst-backup` in the tests that modify `uaclient.conf` ?

Reply:
Yep, both are great ideas, I'll add them